### PR TITLE
upcoming: [M3-8305] - Display Endpoint Type alongside each endpoint hostname in Regions Column & Hostnames Drawers

### PR DESCRIPTION
**Pull request by harsh-akamai**
## Description 📝
As a customer, I want to see a list of endpoints, including the type of each endpoint (E0, E1, E2, E3) alongside each endpoint hostname, in various views related to S3 access keys in Cloud Manager. This will help me understand the endpoints in which each newly created or pre-existing S3 access key can be used.

## Changes  🔄
* Add "Action" Table Header for Actions Column
* Endpoint Type Display in Regions Column
  * Display the type of each endpoint alongside each endpoint hostname in the existing Regions column of the top-level list of both pre-existing and new access keys. 
* (Drawer) Endpoint Type Display for Pre-existing Access Keys
  * Show the type of each endpoint alongside the endpoint hostname when displaying the full list of endpoint hostnames for a specific pre-existing access key.

## Target release date 🗓️
N/A

## Preview 📷

- Access Key Landing Page

| Before  | After   |
| ------- | ------- |
| image|image|

- Access Key Hostnames Drawer

| Before  | After   |
| ------- | ------- |
| image|image|
### Prerequisites
- Test all components for ObjectStorage buckets by toggling the `OBJ Gen2` and the `OBJ Multi-Cluster` feature flags in dev environment

### Reproduction steps
- Go to `serverHandler.ts` file and under the `*object-storage/keys` route, add the `endpoint_type` key-value pair for all `regions` array

### Verification steps
- Turn on MSW
- Navigate to the Access Keys Tab under Object Storage feature
- You should see the last column of the top-level list having a "Actions" header.
- You should find the Endpoint Type alongside each endpoint hostname in the existing Regions column of the top-level list of both pre-existing and new access keys.
- On opening a hostname's Details drawer(only for keys in multiple regions), you should find the endpoint type alongside the endpoint hostname when displaying the full list of endpoint hostnames.

## As an Author I have considered 🤔

*Check all that apply*

- [x] 👀 Doing a self review
- [x] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [ ] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
- [ ] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [ ] 👣 Providing comprehensive reproduction steps
- [x] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

File: packages/manager/src/features/ObjectStorage/AccessKeyLanding/HostNamesDrawer.tsx [hunk 1]
```diff
@@ -30,9 +31,12 @@ export const HostNamesDrawer = (props: Props) => {
...
-                (region) =>
-                  `${regionsLookup[region.id]?.label}: ${region.s3_endpoint}`
+              .map((region) =>
+                region?.endpoint_type
+                  ? `${regionsLookup[region.id]?.label} (${
+                      region.endpoint_type
+                    }): ${region.s3_endpoint}`
+                  : `${regionsLookup[region.id]?.label}: ${region.s3_endpoint}`
               )
               .join('\n') ?? ''
```
Comment (jaalah-akamai) [inline packages/manager/src/features/ObjectStorage/AccessKeyLanding/HostNamesDrawer.tsx:41]: Slightly more readable
```suggestion
              .map((region) => {
                const label = regionsLookup[region.id]?.label;
                const endpointType = region.endpoint_type
                  ? ` (${region.endpoint_type})`
                  : '';
                return `${label}${endpointType}: ${region.s3_endpoint}`;
              })
              .join('\n') ?? ''
```

File: packages/manager/src/features/ObjectStorage/AccessKeyLanding/HostNamesDrawer.tsx [hunk 2]
```diff
@@ -47,11 +51,21 @@ export const HostNamesDrawer = (props: Props) => {
...
+                ? `${region.id} (${region.endpoint_type}): ${region.s3_endpoint}`
+                : `${region.id}: ${region.s3_endpoint}`
+            }
+            value={
+              region?.endpoint_type
+                ? `${regionsLookup[region.id]?.label} (${
+                    region.endpoint_type
+                  }): ${region.s3_endpoint}`
+                : `${regionsLookup[region.id]?.label}: ${region.s3_endpoint}`
+            }
```
Comment (jaalah-akamai) [inline packages/manager/src/features/ObjectStorage/AccessKeyLanding/HostNamesDrawer.tsx:65]: I would do the same as above here. Create some variables for readability

**Review by jaalah-akamai — COMMENTED** [review]
Nice job! Don't forget to add a changeset `yarn changeset` Select manager > upcoming

File: packages/manager/src/features/ObjectStorage/AccessKeyLanding/AccessKeyTable/AccessKeyTableRow.tsx [hunk 3]
```diff
@@ -62,7 +63,15 @@ export const AccessKeyTableRow = ({
...
-      <TableCell>
+      <TableCell
+        sx={{
+          ...(isObjMultiClusterEnabled && {
+            '&&:last-child': {
+              paddingRight: '15px',
+            },
+          }),
+        }}
+      >
```
Comment (coliu-akamai) [inline packages/manager/src/features/ObjectStorage/AccessKeyLanding/AccessKeyTable/AccessKeyTableRow.tsx:74]: nitpick: this `TableCell` and the one in `AccessKeyTable.tsx` both have the same styling applied - could make a shared styled component to use in both if you want

**Review by coliu-akamai — APPROVED** [review]
- ✅  You should see the last column of the top-level list having a "Actions" header.
- ✅  You should find the Endpoint Type alongside each endpoint hostname in the existing Regions column of the top-level list of both pre-existing and new access keys.
- ✅ On opening a hostname's Details drawer(only for keys in multiple regions), you should find the endpoint type alongside the endpoint hostname when displaying the full list of endpoint hostnames.

nice work! left one small, non-blocking optional comment

**Review by coliu-akamai — APPROVED** [review]

**Review by jaalah-akamai — APPROVED** [review]

**Pull request merged**